Fake quantization enhancements for QAT/PTQ support- fix tests

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26516 Per-channel baseline
* #26520 control of observer/fake-quant operations
* #26622 Default histogram observer
* #26519 Serialization and range reduction support for Fake Quant/Observer
* **#26876 Fake quantization enhancements for QAT/PTQ support- fix tests**

Add ability to turn fake quantization and observers independently.

Differential Revision: [D17592961](https://our.internmc.facebook.com/intern/diff/D17592961/)

raghuramank10000 committed:
Fake quantization enhancements for QAT/PTQ support- fix tests

Add ability to turn fake quantization and observers independently.

Differential Revision: [D17592961](https://our.internmc.facebook.com/intern/diff/D17592961/)

[ghstack-poisoned]

dzhulgakov commented on torch/nn/_intrinsic/qat/modules/conv_fused.py:
```diff
@@ -80,13 +80,13 @@ def reset_parameters(self):
     def enable_fake_quant(self):
-        self.observer.enable()
-        self.weight_fake_quant.enable()
+        self.observer.enable_fake_quant()
```
I feel the ordering of the PRs is weird as you delete it later :) but it's ok

raghuramank100 replied:
Yes, on hind-sight having fewer PRs would have been better.

dzhulgakov reviewed (approved):
Tests on this one fail with autograd error which seems to be related. But it's probably from another PR in the stack

raghuramank10000 committed:
Update on "Fake quantization enhancements for QAT/PTQ support- fix tests"


Add ability to turn fake quantization and observers independently.

Differential Revision: [D17592961](https://our.internmc.facebook.com/intern/diff/D17592961/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Fake quantization enhancements for QAT/PTQ support- fix tests"


Add ability to turn fake quantization and observers independently.

Differential Revision: [D17592961](https://our.internmc.facebook.com/intern/diff/D17592961/)

[ghstack-poisoned]
